### ISSUE TITLE
BSRCH (Bloomberg Search)

### ISSUE THREAD

**Issue by MarekOzana** [Rblp/Rblpapi#82]
Hi,
Would somebody have time to implement BSRCH function? Example us case
BSRCH("COMDTY:NGFLOW")
It is commonly used for Fixed Income instrument screening.
 Thank you for your time and help.
    /Marek

**Comment by eddelbuettel** [comment]
Well, considering that
- I do not know what API function implements it
- A casual look at the [developer guide](http://www.bloomberglabs.com/content/uploads/sites/2/2014/07/blpapi-developers-guide-2.54.pdf) does not reveal anything either
- We generally follow open source contribution principles: "he who has an itch gets to scratch it"

We are all busy professionals too, and hence unlikely to write this for you.  If you want to take stab, we can try to help.  Or you can try to hire someone and we will try to help to get that work integrated.

**Comment by joel23888** [comment]
It seems clear that the functionality in the BSRCH Excel function is not supported by the BBG C++ API. I had a look through the developer's guide and cannot see how this would be possible.

**Comment by eddelbuettel** [comment]
Ahhh. Good find. Closing this as a consequence of that comment.

**Comment by n-e-w** [comment]
Incidentally, you can work around similar queries to extract elements from
known datasets by using some `bds` calls (though calling BSRCH is only
available if you're a Bloomberg employee -- at least when I tried to run on
my terminal?).

For example, let's say you want the top 10 holdings of an ETF -- a field in
the API which is "TOP_MUTUAL_FUND_HOLDINGS" --  you can call:

    con <- Rblpapi::blpConnect()
    Rblpapi::bds("SPY US Equity", "TOP MUTUAL FUND HOLDINGS")

This will return a `data.frame` of the top 10 holdings. You can then search
this data frame for the ticker or name of the security depending on what
you have to start with.

Could put this in a wrapper pretty easily if you know what `bds` dataset
your security *name* (rather than ticker) might appear in and then extract
it.

Just a thought.


On Thu, Dec 10, 2015 at 1:14 PM, Dirk Eddelbuettel <notifications@github.com

**Comment by wmorgan85** [comment]
Basic BSRCH functionality has now been implemented in #111.

**Comment by eddelbuettel** [comment]
And as a courtesy I uploaded _source_ and _Windows release aka 3.2.*_ packages to the [ghrr drat](https://github.com/ghrr/drat).  The sequence

```r
if (!require("drat")) install.packages("drat")     ## install if needed
drat::addRepo("ghrr")
install.packages("Rblpapi")                           
```

will install it.   It may be a moment before this gets to CRAN so this provides a preview.
